From dbb1c2d10c127b9a5f99e3e4ee5981f5222fc7e7 Mon Sep 17 00:00:00 2001
From: Wayne Davison <wayne@opencoder.net>
Date: Sun, 4 Jul 2021 12:15:16 -0700
Subject: [PATCH] Set whole_file = 0 when whole_file < 0.  Fixes issue 114.

---
 receiver.c | 3 +++
 sender.c   | 4 ++++
 2 files changed, 7 insertions(+)

diff --git a/receiver.c b/receiver.c
index 9c47ece3..e85c4779 100644
--- a/receiver.c
+++ b/receiver.c
@@ -539,6 +539,9 @@ int recv_files(int f_in, int f_out, char *local_name)
 	if (delay_updates)
 		delayed_bits = bitbag_create(cur_flist->used + 1);
 
+	if (whole_file < 0)
+		whole_file = 0;
+
 	progress_init();
 
 	while (1) {
diff --git a/sender.c b/sender.c
index 9cfca134..83603b99 100644
--- a/sender.c
+++ b/sender.c
@@ -35,6 +35,7 @@ extern int append_mode;
 extern int copy_links;
 extern int io_error;
 extern int flist_eof;
+extern int whole_file;
 extern int allowed_lull;
 extern int preserve_xattrs;
 extern int protocol_version;
@@ -204,6 +205,9 @@ void send_files(int f_in, int f_out)
 	if (DEBUG_GTE(SEND, 1))
 		rprintf(FINFO, "send_files starting\n");
 
+	if (whole_file < 0)
+		whole_file = 0;
+
 	progress_init();
 
 	while (1) {
